### DISCUSSION TITLE
Is it possible to use both Ros and Ros2 at the same time?

### DISCUSSION THREAD

**Discussion by Kevin-Bobby** [Yaskawa-Global/motoros2#261] [Q&A] [answered]
I have lots of scripts written to perform various tasks using ROS and I am currently using Ros to controll my robot(GP12), but since Ros is reaching EOL, i was wondering if it is possible to control the robot through ros and ros2 preferably through different systems and not simultaneously, to test Ros2 code while continuing to do required work using ROS

**Comment by ted-miller** [comment, marked as answer]
Hi Kevin-Bobby,
I have not tested the combination of MotoROS and MotoROS2 installed on the same robot controller. So, I cannot say with certainty.

However, I expect that there would be some resource conflicts between the two application. There is a lot of re-used code in MotoROS2. I doubt that they can coexist.
